server/src/main/java/org/elasticsearch/index/mapper/NestedObjectMapper.java

```diff
@@ -137,6 +139,16 @@ public void setIncludeInRoot(boolean includeInRoot) {
         return Collections.unmodifiableMap(this.mappers);
     }
 
+    @Override
+    public ObjectMapper.Builder newBuilder() {
```

Comment:
Maybe pass `indexVersionCreated` in rather than make the extra member. It looks like it's pretty close at hand on the caller.